### PR TITLE
os/bluestore: remove incorrectly comments.

### PR DESCRIPTION
**Pull request by majianpeng**
Signed-off-by: Jianpeng Ma <jianpeng.ma@intel.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/os/bluestore/BlueFS.cc [hunk 1]
```diff
@@ -1881,8 +1881,9 @@ int BlueFS::_read_random(
   FileReader *h,         ///< [in] read from here
   uint64_t off,          ///< [in] offset
   uint64_t len,          ///< [in] this many bytes
-  char *out)             ///< [out] optional: or copy it here
+  char *out)
 {
+  ceph_assert(out);
```
Comment (tchaikov) [inline src/os/bluestore/BlueFS.cc]: i'd avoid putting `ceph_assert()` in the critical path. moreover, probably we can remove the check for null couple lines below:
```c++
      if (out) {
        // NOTE: h->bl is normally a contiguous buffer so c_str() is free.
        memcpy(out, buf->bl.c_str() + off - buf->bl_off, r);
        out += r;
      }
```

**Comment by majianpeng** [comment]
@tchaikov . update by your comments.

File: src/os/bluestore/BlueFS.cc [hunk 2]
```diff
@@ -1881,7 +1881,7 @@ int BlueFS::_read_random(
   FileReader *h,         ///< [in] read from here
   uint64_t off,          ///< [in] offset
   uint64_t len,          ///< [in] this many bytes
-  char *out)             ///< [out] optional: or copy it here
+  char *out)
```
Comment (tchaikov) [inline src/os/bluestore/BlueFS.cc]: instead of removing the comment, we can improve it like
```
char *out)             ///< [out]: copy it here
```

**Comment by majianpeng** [comment]
update.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2020-03-26_11:37:47-rados-wip-kefu-testing-2020-03-26-1712-distro-basic-smithi/

**Pull request merged**